### PR TITLE
Run e2e locally

### PR DESCRIPTION
**Pull request by alvaroaleman**
**What this PR does / why we need it**:

This makes the e2e tests work from localhost without having to copy binaries over

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: test/e2e/provisioning/helper.go [hunk 1]
```diff
@@ -127,28 +126,27 @@ func runScenarios(t *testing.T, excludeSelector *scenarioSelector, testParams st
 }
 
 func runVerifyTool(manifestPath string, params string) error {
-	homeDir := os.Getenv("HOME")
-	kubeConfigDir := filepath.Join(homeDir, ".kube/config")
+	gopath := os.Getenv("GOPATH")
```
Comment (p0lyn0mial) [inline test/e2e/provisioning/helper.go]: nit: usually home dir is the default path for `kubeconfig`.

File: test/e2e/provisioning/helper.go [hunk 2]
```diff
@@ -127,28 +126,27 @@ func runScenarios(t *testing.T, excludeSelector *scenarioSelector, testParams st
...
 		"true",
 	}
 
-	cmd := exec.Command("../../tools/verify/verify", args...)
-	var out bytes.Buffer
-	cmd.Stderr = &out
-
-	err := cmd.Run()
+	cmd := exec.Command(filepath.Join(projectDir, "test/tools/verify/verify"), args...)
+	out, err := cmd.CombinedOutput()
```
Comment (p0lyn0mial) [inline test/e2e/provisioning/helper.go]: this is elegant :)

File: test/tools/integration/provision_master.sh [hunk 3]
```diff
@@ -17,7 +17,7 @@ done
 
 
 rsync -av  -e "ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no" \
-    ../../../{examples,machine-controller,Dockerfile,test} ../verify/verify \
+    ../../../{examples,machine-controller,Dockerfile,test/tools/verify/verify} \
```
[the remark this answers is not in the document]
Comment (p0lyn0mial) [reply test/tools/integration/provision_master.sh]: also do we have to copy `examples` directory ?

File: test/e2e/provisioning/helper.go (hunk 1 above)
Comment (alvaroaleman) [reply test/e2e/provisioning/helper.go]: Yeah that is true, but I wanted to allow to run it locally and do not want it to overwrite my kubeconfig

Comment (p0lyn0mial) [reply test/e2e/provisioning/helper.go]: how can I run the tests from my local machine ?

Comment (alvaroaleman) [reply test/e2e/provisioning/helper.go]: If you set all env variables, run the `provision_master.sh` once manually you can just execute `go test -tags=e2e -parallel 24 -v -timeout 10m  ./test/e2e/...`

I could probably polish it a little bit more and write a maketarget that does both

**Pull request merged**